src/frontend/js/forms/update-permalink.ts

```diff
@@ -16,12 +26,53 @@ window.addEventListener("load", () => {
         const url = (document.querySelector('[for="id_title"]') as HTMLElement)
           .dataset.slugifyUrl;
         slugify(url, { title: currentTitle }).then((response) => {
-          document
-            .getElementById("id_slug")
-            .setAttribute("value", response["unique_slug"]);
+          /* on success write response to slug field and adjust the fields size */
+          slugField.value = response["unique_slug"];
+          slugField.size = response["unique_slug"].length;
         });
       });
   }
+
+  function replaceButtonIcon(buttonIcon: string, toReplace: SVGElement) {
+    let newButtonIcon = document.createElement("i");
+    newButtonIcon.setAttribute("data-feather", buttonIcon);
+    toReplace.replaceWith(newButtonIcon);
+  }
+
+  editButton.addEventListener("click", function(e){
+      if (slugField.readOnly === true) {
+          // make slug field editable and buffer current value
+          currentSlug = slugField.value;
+          replaceButtonIcon("save", editButton.querySelector("svg"));
+          replaceButtonIcon("x-circle", copyButton.querySelector("svg"));
+          slugField.readOnly = false;
+      } else {
+          // save edited slug field, but only in the input field, which gets persistent at form submission
+          replaceButtonIcon("edit-3", editButton.querySelector("svg"));
+          replaceButtonIcon("copy", copyButton.querySelector("svg"));
+          slugField.readOnly = true;
+      }
+      slugDiv.classList.toggle("editable");
+      feather.replace();
+  });
+
+  copyButton.addEventListener("click", function(e){
+      if (slugField.readOnly === true) {
+          // copy whole permalink to clipboard
+          let slugContainer = document.getElementById("link-container");
+          let baseUrl = slugContainer.querySelector("span").innerHTML;
+          let slugValue = slugContainer.querySelector("input").value;
+          navigator.clipboard.writeText(baseUrl.concat(slugValue));
+      } else {
+          // reset slug field to read-only and restore slug value
+          slugField.value = currentSlug;
+          replaceButtonIcon("edit-3", editButton.querySelector("svg"));
+          replaceButtonIcon("copy", copyButton.querySelector("svg"));
+          slugField.readOnly = true;
+          slugDiv.classList.toggle("editable");
+          feather.replace();
+      }
+  });
```

Comment:
I would not use the same buttons for different actions. I think this makes it hard to maintain the source code in the future. Just use separate buttons and add the class "hidden" to the buttons which should not be displayed at the moment.

Reply:
I have to admit, that this "double functional" buttons caused some headache while coding :sweat_smile: 
As they have not really common properties except for their position in the form, we can of course donate two more buttons